.github/workflows/run-unit-tests.yml

```diff
@@ -0,0 +1,13 @@
+name: Unit Testing
```

Comment:
Nice can you add linting too?

Reply:
yeah I was in the middle of creating a separate branch for that but I can try to add it here